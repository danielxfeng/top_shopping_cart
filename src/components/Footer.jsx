const Footer = () => {
  return (
    <footer>
      <p>
        <span>&copy; 2024</span>
        <a
          href="https://github.com/danielxfeng/top_shopping_cart"
          target="_blank"
          rel="noopener noreferrer"
        >
          Fancy Mall
        </a>
      </p>
      <p>
        Made by ❤️<em>Daniel</em>❤️
      </p>
    </footer>
  );
};

export default Footer;
