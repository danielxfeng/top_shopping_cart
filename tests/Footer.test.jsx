import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Footer from "../src/components/Footer";

describe("Footer page", () => {
  it("renders correct", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("renders the content1", () => {
    render(<Footer />);
    expect(screen.getByText(/Â© 2024/i)).toBeInTheDocument();
  });

  it("renders the content2", () => {
    render(<Footer />);
    expect(screen.getByText(/Daniel/i)).toBeInTheDocument();
  });

  it("renders an external link with correct href", () => {
    render(<Footer />);
    const linkElement = screen.getByText("Fancy Mall");
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute("href", "https://github.com/danielxfeng/top_shopping_cart");
    expect(linkElement).toHaveAttribute("target", "_blank");
    expect(linkElement).toHaveAttribute("rel", "noopener noreferrer");
  });
});
