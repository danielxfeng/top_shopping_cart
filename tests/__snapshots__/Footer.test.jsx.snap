// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer page > renders correct 1`] = `
<div>
  <footer
    class="_footer_4d8704"
  >
    <p>
      <span>
        © 2024
      </span>
      <a
        href="https://github.com/danielxfeng/top_shopping_cart"
        rel="noopener noreferrer"
        target="_blank"
      >
        Fancy Mall
      </a>
    </p>
    <p>
      Made by ❤️
      <em>
        Daniel
      </em>
      ❤️
    </p>
  </footer>
</div>
`;
